api/src/test/java/org/openmrs/api/impl/ConditionServiceImplTest.java

```diff
@@ -0,0 +1,172 @@
+		condition.setCondition(codedOrFreeText);
+		condition.setClinicalStatus(ConditionClinicalStatus.ACTIVE);
+		condition.setUuid(uuid);
+		condition.setPatient(patientService.getPatient(2));
+		conditionService.saveCondition(condition);
+		Condition savedCondition = conditionService.getConditionByUuid(uuid);
+		Assert.assertEquals(patientId, savedCondition.getPatient().getPatientId());
+		Assert.assertEquals(uuid, savedCondition.getUuid());
+		Assert.assertEquals(codedOrFreeText, savedCondition.getCondition());
+		Assert.assertEquals(ConditionClinicalStatus.ACTIVE, savedCondition.getClinicalStatus());
+		Assert.assertNotNull(savedCondition.getConditionId());
+	}
+
+	/**
+	 * @see ConditionService#getConditionByUuid(String)
+	 */
+	@Test
+	public void getConditionByUuid_shouldFindConditionGivenValidUuid() {
+		String uuid="2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		Condition condition = conditionService.getConditionByUuid(uuid);
+		Assert.assertEquals(uuid, condition.getUuid());
+	}
+
+	/**
+	 * @see ConditionService#getConditionByUuid(String)
+	 */
+	@Test
+	public void getConditionByUuid_shouldReturnNullIfNoObjectFoundWithGivenUuid() {
+		Assert.assertNull(conditionService.getConditionByUuid("invalid uuid"));
+	}
+
+	/**
+	 * @see ConditionService#getCondition(Integer) 
+	 */
+	@Test
+	public void getCondition_shouldFindConditionGivenValidId() {
+		Condition condition = conditionService.getCondition(1);
+		Assert.assertEquals(ConditionClinicalStatus.INACTIVE, condition.getClinicalStatus());
+		Assert.assertEquals(ConditionVerificationStatus.PROVISIONAL, condition.getVerificationStatus());
+		Assert.assertEquals("2cc6880e-2c46-15e4-9038-a6c5e4d22fb7", condition.getUuid());
+	}
+
+	/**
+	 * @see ConditionService#getActiveConditions(Patient)
+	 */
+	@Test
+	public void getActiveConditions_shouldGetActiveConditions() {
+		List<Condition> activeConditions = conditionService.getActiveConditions(patientService.getPatient(2));
+		Condition firstActiveCondition = activeConditions.get(0);
+		Condition secondActiveCondition = activeConditions.get(1);
+		String firstUUID = "2cc6880e-2c46-11e4-9138-a6c5e4d20fb7";
+		String secondUUID = "2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		Assert.assertTrue(activeConditions.size() == 2);
+		Assert.assertEquals(firstUUID,firstActiveCondition.getUuid());
+		Assert.assertEquals(secondUUID, secondActiveCondition.getUuid());
+	}
+	
+	
+
+	/**
+	 * @see ConditionService#voidCondition(Condition, String) 
+	 */
+	@Test
+	public void voidCondition_shouldVoidConditionSuccessfully(){
+		String voidReason = "Test Reason";
+		Condition nonVoidedCondition = conditionService.getCondition(2);
+		Assert.assertFalse(nonVoidedCondition.getVoided());
+		Assert.assertNull(nonVoidedCondition.getVoidReason());
+		Assert.assertNull(nonVoidedCondition.getDateVoided());
+		Assert.assertNull(nonVoidedCondition.getVoidedBy());
+		
+		Condition voidedCondition = conditionService.voidCondition(nonVoidedCondition, voidReason);
```

Comment:
After voiding, load the condition afresh using conditionService.getCondition()

Reply:
Okay. Done